### PR TITLE
Cookie based page_mode locking

### PR DESCRIPTION
**Pull request by philbogle**
Allow a "sticky" page_mode to be specified for a session by hitting (e.g.)  

http://localhost.code.org:3000/private/pm/starwars

replacing starwars with whatever pagemode you want to set for the session.

The page mode is saved in a session cookie and applies to all subsequent requests in the session.

Caveat: some of our pages are cached on the client, so you may need to refresh the page or clear your cache to see the changes.

The previous approach of specifying the page_mode via a query string parameter on any request is no longer supported.  The problem is that it didn't work in all cases; e.g. for pages like teacher-dashboard that fetched part of their content using Ajax requests which failed to pass along the parameter.

### CODE REVIEW

File: pegasus/router.rb [hunk 1]
```diff
@@ -145,6 +145,13 @@ def self.set_max_age(type, default)
     redirect "/#{path}"
   end
 
+  # Page mode selection
+  get '/private/pm/*' do |page_mode|
+    dont_cache
+    response.set_cookie('pm', {value: page_mode, domain: ".#{request.site}", path: '/', expires: Time.now + 3600*4})
```
Comment (breville) [inline pegasus/router.rb]: this looks like a 4 hour cookie?  there is also a session cookie, in which expires is not set, if we preferred that..

**Comment by breville** [comment]
apart from the session cookie question, and my lack of experience with varnish etc., this seems okay

**Comment by philbogle** [comment]
The tests showed an issue with http_cache config changes, fixing.

**Comment by philbogle** [comment]
Fixed the http_cache test issue.

**Comment by philbogle** [comment]
The LANGUAGE_COOKIES constant in http_cache isn't very accurately named (since it now includes the page mode cookie) and there is some repetition oacross LANGUAGE_COOKIES and whitelist_cookies that could be factored out.  I want to keep this change minimal, but will fix this on Monday.

**Pull request merged**